gateway/handle.go

```diff
@@ -819,6 +830,15 @@ func (gw *Handle) extractJobsFromInternalBatchPayload(reqType string, body []byt
 	return jobs, nil
 }
 
+func (gw *Handle) getSourceConfigFromSourceID(sourceID string) backendconfig.SourceT {
+	gw.configSubscriberLock.RLock()
+	defer gw.configSubscriberLock.RUnlock()
+	if s, ok := gw.sourceIDSourceMap[sourceID]; ok {
+		return s
+	}
+	return backendconfig.SourceT{}
+}
```

Comment:
I feel instead of returning too much information, we should just return, what we want (writeKey). wdyt?
```suggestion
func (gw *Handle) getWriteKeyFromSourceID(sourceID string) (string, bool) {
	gw.configSubscriberLock.RLock()
	defer gw.configSubscriberLock.RUnlock()
	if s, ok := gw.sourceIDSourceMap[sourceID]; ok {
		return s.WriteKey, true
	}
	return "", false
}
```

Reply:
I understand, we changed the signature for newSourceStatTagsWithReason which `backendconfig.SourceT` instead of `*gwtypes.AuthRequestContext`. But I didn't understand why it was done.